src/group_theory/nilpotent.lean

```diff
@@ -675,3 +685,53 @@ begin
   rw [eq_bot_iff, ←hn],
   exact derived_le_lower_central n,
 end
+
+section classical
+
+open_locale classical
+
+/-- A custom induction principle fintypes. The base case is a subsingleton type,
+and the induction step is for non-trivial types, and one can assume the hypothesis for
+smaller groups (via `fintype.card`).
+-/
+@[elab_as_eliminator]
+lemma fintype.induction_subsingleton_or_nontrivial
+  {P : Π α [fintype α], Prop}
+  (α : Type*)
+  [fintype α]
+  (hbase : ∀ α [fintype α] [subsingleton α], by exactI P α)
+  (hstep : ∀ α [fintype α] [nontrivial α],
+    by exactI ∀ (ih : ∀ β [fintype β], by exactI ∀ (h : fintype.card β < fintype.card α), P β),
+    P α) :
+  P α :=
+begin
```

Comment:
I have a tendency to factor our the induction argument into separate lemmas, but I am not sure how useful and welcome that is here.

Is this a useful induction principle to add to `fintyp`? If not I can fold it back into `is_nilpotent.of_p_group`.

Reply:
I think this one is useful to have! Of course, in the final stages of this PR, it should be moved to a file on fintypes.